[DOCS] Refine quickstart docs further for 1.0

lokeshj1703 opened the pull request:
### Change Logs

Adds more quickstart docs around indexing, partial updates and merge mode.
Further refines the indexing docs, adds limitations and index acceleration related information.

### Impact

Refines quickstart docs 

### Risk level (write none, low medium or high below)

low

### Documentation Update

NA

### Contributor's checklist

- [ ] Read through [contributor's guide](https://hudi.apache.org/contribute/how-to-contribute)
- [ ] Change Logs and Impact were stated clearly
- [ ] Adequate tests were added if applicable
- [ ] CI passed

codope commented on website/docs/metadata_indexing.md:
```diff
@@ -11,27 +11,30 @@ of Hudi depends on the metadata table. Different types of index, from `files` in
-created in two phases and uses a mix of optimistic concurrency control and log-based concurrency control models. The two
+can take hours to index. This is where Hudi's novel asynchronous metadata indexing comes into play. 
+
+## Asynchronous Metadata Indexing
```
Let's call it concurrent indexing?

lokeshj1703 replied:
Addressed

codope commented on website/docs/sql_ddl.md:
```diff
@@ -283,13 +322,7 @@ DROP INDEX [IF EXISTS] index_name ON [TABLE] table_name
-:::
-
-#### Create Functional Index (Experimental)
+#### Create Functional Index
```
let's make it Expression Index and add the sql and functions supported here.

lokeshj1703 replied:
Addressed

lokeshj1703 commented on website/docs/metadata.md:
````diff
@@ -61,6 +62,11 @@ Following are the different types of metadata currently supported.
     hoodie.metadata.index.column.stats.columns=<comma-separated-column-names>
   ```
 
+- ***Secondary Index***
````
@codope @vc I have added Secondary Index here. I see that Expression index, record index, bloom filter index is missing on this page but are covered in `website/docs/indexes.md`.
Should we rather refer to indexes page in this doc?

vinothchandar replied:
I want this page to be about metadata (col stats, file listings, partition stats). lets do indexes on the indexes page

lokeshj1703 replied:
Reverted this change.

vinothchandar commented on website/docs/sql_ddl.md:
```diff
@@ -342,65 +377,65 @@ CREATE TABLE hudi_table_func_index (
 </details>
 
-#### Create Partition Stats and Secondary Index (Experimental)
+#### Create Partition Stats and Secondary Index
```
do we need to set hoodie.metadata.enable and hoodie.eable.data.skipping anymore? are nt they defaults..

vinothchandar replied:
we should simplify this down as much as possible..

codope replied:
You're right. Now those two need not be set, as they are true by default.

vinothchandar reviewed (commented):
Overall, can you please confirm if we can trim down some extra configs (which are not needed due to changed defaults) and how we use record mergers and not payloads.

codope commented on website/docs/sql_ddl.md:
```diff
@@ -473,8 +508,10 @@ Time taken: 0.83 seconds, Fetched 2 row(s)
   skipping. Queries with such predicates cannot leverage the indexes.
 - Secondary index is not supported for nested fields.
+- Secondary index can be created only if record index is available in the table
+- Secondary index can only be used for tables using OverwriteWithLatestAvroPayload payload or COMMIT_TIME_ORDERING merge mode 
```
Yeah we can say, secondary index is not yet supported with EVENT_TIME and CUSTOM merge modes.

codope commented on website/docs/sql_ddl.md:
```diff
@@ -103,6 +103,49 @@ TBLPROPERTIES (
+  primaryKey = 'id',
+  precombineField = 'ts',
+  recordMergeMode = 'CUSTOM',
+  'hoodie.datasource.write.payload.class' = 'org.apache.hudi.common.model.PartialUpdateAvroPayload'
```
ok i will update with the merger example.

codope commented on website/docs/quick-start-guide.md:
```diff
@@ -454,27 +454,47 @@ VALUES
 -- Create secondary index on rider column
 CREATE INDEX record_index ON hudi_indexed_table (uuid);
 CREATE INDEX idx_rider ON hudi_indexed_table (rider);
+SET hoodie.metadata.record.index.enable=true;
```
yes

lokeshj1703 commented on website/docs/quick-start-guide.md:
```diff
@@ -454,27 +454,47 @@ VALUES
 
--- Create bloom filter expression index on city column
-CREATE INDEX idx_bloom_city ON hudi_indexed_table USING bloom_filters(city) OPTIONS(expr='identity');
+-- Create bloom filter expression index on driver column
```
I have changed the config to use `hoodie.write.record.merge.mode` which looks better.
```
hoodie.write.record.merge.mode = "COMMIT_TIME_ORDERING"
```

lokeshj1703 commented:
>Overall, can you please confirm if we can trim down some extra configs (which are not needed due to changed defaults) and how we use record mergers and not payloads.

Addressed both. Removed `hoodie.enable.data.skipping=true` usage and `hoodie.metadata.enable=true` usage. Also using record merge mode instead of payload now for secondary index.

vinothchandar reviewed (commented):
I'll make another pass
